src/rgw/rgw_rest_s3.cc

```diff
@@ -4420,3 +4367,117 @@ std::string RGWLDAPTokenExtractor::get_token() const
   }
   return token;
 }
+
+bool RGWS3V2TokenBasedAuthEngine::is_applicable() const noexcept
+{
+  if (!RGWTokenBasedAuthEngine::is_applicable()) {
+    return false;
+  }
+
+  if (!date.empty()) {
+    time_t exp = atoll(date.c_str());
+    if (now >= exp) {
+      return false;
```

Comment:
Are you sure that only skipping a given auth engine when `Expires` is in the past  is the intended behavior? I saw that `RGW_Auth_S3::authorize_v2` signalizes `EPERM` in such case.